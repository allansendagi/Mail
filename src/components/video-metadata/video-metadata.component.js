import React from 'react';
import { Button, Divider, Icon } from 'semantic-ui-react';
import {Rating} from '../rating/rating.component';
import './video-metadata.styles.scss';

export function VideoMetadata(props) {
	const viewcount = Number(props.viewcount).toLocaleString() || '';
	return(
		<div className='video-metadata'>
		 <h3> Video title</h3>
		 <div className='video-stats'>
		   <span>{viewcount} views</span>
		   <div className='video-actions'>
		    <Rating likeCount={1000} dislikeCount={100} />
		     <Button basic icon labelPosition='left'>
		       <Icon name='share'/>
		        share
		        </Button>
		        <Button basic icon>
              <Icon name='add circle' />
             </Button>
             <Button basic icon>
            <Icon name='ellipsis horizontal' />
          </Button>
		   </div>
		 </div>
		<Divider/>	
		</div>
	);
}