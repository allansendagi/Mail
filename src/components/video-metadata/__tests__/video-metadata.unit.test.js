import React from 'react';
import {shallow, configure} from 'enzyme';
import {VideoMetadata} from '../video-metadata.component';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});
describe('VideoMetadata', () => {
  test('renders without props', () => {
    const wrapper = shallow(<VideoMetadata/>);
    expect(wrapper).toMatchSnapshot();
  });
  test('renders with view count', () => {
    const wrapper = shallow(<VideoMetadata viewCount={100234}/>);
    expect(wrapper).toMatchSnapshot();
  });
});