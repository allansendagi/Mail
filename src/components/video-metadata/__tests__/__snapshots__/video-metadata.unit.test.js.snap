// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoMetadata renders with view count 1`] = `ShallowWrapper {}`;

exports[`VideoMetadata renders without props 1`] = `ShallowWrapper {}`;
